### PR TITLE
feat(new-widget-builder-experience): Add Overwrite Widget modal

### PR DESCRIPTION
**Pull request by narsaynorath**
When a user selects a widget card from the sidebar, we should inform them that this will overwrite their current values and they should confirm the action.

Screen Shot 2022-02-28 at 4 29 36 PM

I refactored the hovering styles out of the `card.tsx` component to make it more simple since the modal doesn't need to pass handlers or anything. The responsibility is currently shifted to the parent to define a wrapper to suit its use case.

Note: The modal currently appears for every selection, I will follow up with another PR to trigger the modal only if the form is dirty.

### CODE REVIEW

File: tests/js/spec/components/modals/widgetBuilderOverwriteModal.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        closeModal={() => undefined}
+        iconColor="white"
+        widget={widget}
+        onConfirm={jest.fn()}
+      />
+    );
+
+    expect(screen.getByText('Overwrite Widget')).toBeVisible();
+    expect(screen.getByText('Test title')).toBeVisible();
+    expect(screen.getByText('Test description')).toBeVisible();
```
Comment (matejminar) [inline tests/js/spec/components/modals/widgetBuilderOverwriteModal.spec.tsx:33]:
```suggestion
    expect(screen.getByText('Overwrite Widget')).toBeInTheDocument();
    expect(screen.getByText('Test title')).toBeInTheDocument();
    expect(screen.getByText('Test description')).toBeInTheDocument();
```

**Review by matejminar — APPROVED** [review]

File: static/app/components/modals/widgetBuilder/overwriteWidgetModal.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        <h4>{t('Overwrite Widget')}</h4>
+      </Header>
+      <Body>
+        {MODAL_DESCRIPTION}
+        <CardWrapper>
+          <Card widget={widget} iconColor={iconColor} />
+        </CardWrapper>
+      </Body>
+      <Footer>
+        <Actions>
```
Comment (priscilawebdev) [inline static/app/components/modals/widgetBuilder/overwriteWidgetModal.tsx:51]:
```suggestion
        <ButtonBar gap={1.5}>
```

**Review by priscilawebdev — APPROVED** [review]
left some suggestions but it looks good to me

File: static/app/components/modals/widgetBuilder/overwriteWidgetModal.tsx (hunk 2 above)
Comment (narsaynorath) [reply static/app/components/modals/widgetBuilder/overwriteWidgetModal.tsx]: Awesome! thanks

File: static/app/actionCreators/modal.tsx [hunk 3]
```diff
@@ -241,6 +242,15 @@ export async function openAddDashboardWidgetModal(options: DashboardWidgetModalO
   openModal(deps => <Modal {...deps} {...options} />, {backdrop: 'static', modalCss});
 }
 
+export async function openWidgetBuilderOverwriteModal(
```
[the remark this answers is not in the document]
Comment (narsaynorath) [reply static/app/actionCreators/modal.tsx]: 👍 Moved this open function into the widget library sidebar file, but I kept the modal component and test file where they are

**Pull request merged**